sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/FilterAnalyzer.scala

```diff
@@ -0,0 +1,186 @@
+        list.append((pName, value))
+        true
+
+      case GreaterThan(attr, value) =>
+        queryBuilder.append(canonicalCosmosFieldPath(attr)).append(">").append(pName)
+        list.append((pName, value))
+        true
+
+      case LessThanOrEqual(attr, value) =>
+        queryBuilder.append(canonicalCosmosFieldPath(attr)).append("<=").append(pName)
+        list.append((pName, value))
+        true
+
+      case GreaterThanOrEqual(attr, value) =>
+        queryBuilder.append(canonicalCosmosFieldPath(attr)).append(">=").append(pName)
+        list.append((pName, value))
+        true
+
+      case In(attr, values) =>
+        queryBuilder.append(canonicalCosmosFieldPath(attr)).append(" IN ")
+        queryBuilder.append("(")
+        queryBuilder.append(
+          values.map(
+            value => {
+              val pName = paramName(list.size)
+              list.append((pName, value))
+              pName
+            }
+          ).mkString(","))
+        queryBuilder.append(")")
+        true
+
+      case StringStartsWith(attr, value: String) =>
+        queryBuilder.append("STARTSWITH(").append(canonicalCosmosFieldPath(attr)).append(pName).append(")")
+        list.append((pName, value))
+        true
+
+      case StringEndsWith(attr, value: String) =>
+        queryBuilder.append("ENDSWITH(").append(canonicalCosmosFieldPath(attr)).append(pName).append(")")
+        list.append((pName, value))
+        true
+
+      case StringContains(attr, value: String) =>
+        queryBuilder.append("CONTAINS(").append(canonicalCosmosFieldPath(attr)).append(pName).append(")")
+        list.append((pName, value))
+        true
+
+      case IsNull(attr) =>
+        queryBuilder.append(s"IS_NULL(${canonicalCosmosFieldPath(attr)})")
+        true
+
+      case IsNotNull(attr) =>
+        queryBuilder.append(s"NOT(IS_NULL(${canonicalCosmosFieldPath(attr)}))")
+        true
+
+      case And(leftFilter, rightFilter) =>
+        queryBuilder.append("(")
+        val isLeftCosmosPredicate = appendCosmosQueryPredicate(queryBuilder, list, leftFilter)
+        queryBuilder.append(" AND ")
+        val isRightCosmosPredicate = appendCosmosQueryPredicate(queryBuilder, list, rightFilter)
+        queryBuilder.append(")")
+        isLeftCosmosPredicate && isRightCosmosPredicate
+
+      case Or(leftFilter, rightFilter) =>
+        queryBuilder.append("(")
+        val isLeftCosmosPredicate = appendCosmosQueryPredicate(queryBuilder, list, leftFilter)
+        queryBuilder.append(" OR ")
+        val isRightCosmosPredicate = appendCosmosQueryPredicate(queryBuilder, list, rightFilter)
+        queryBuilder.append(")")
+        isLeftCosmosPredicate && isRightCosmosPredicate
+
+      case Not(childFilter) =>
+        queryBuilder.append("NOT(")
+        val isInnerCosmosPredicate = appendCosmosQueryPredicate(queryBuilder, list, childFilter)
+        queryBuilder.append(")")
+        isInnerCosmosPredicate
+
+      case _: Filter =>
```

Comment:
As a ToDo I would also track adding some of the aggregates under certain conditions:
Count, Avg, Min, Max - We need to double-check when they can be served from full fidleity index - I think (but needs to be re-validated) when either no Group By is used or exactly one Group By by logical partition key - anyway would be worthwile to have a conversation with the query team about.

Reply:
Same is true for certain Order By statements - like simple query with just order by by columns supported in index - it is possible that it simply isn't worth doing this because Spark is efficient at ordering and it is too complicated to parse and decide to only push down when order by combination matches supporting indexes. But worth having a separate discussion about and making an intentional call- and adding unit test coverage to ensure we comply with the intentional decision long-term.

Reply:
These spark Filter are just predicate. I. don't see the support for count, avg, etc in the Spark Filter.
I suspect they will be provided from a different path. Need to investigate on that. 

for now, I added a TODO to look at where those will be pushed down.